x-pack/plugins/security_solution_serverless/server/cloud_security/tasks/usage_reporting_task.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Response } from 'node-fetch';
+
+import type { CoreSetup, ElasticsearchClient, Logger, LoggerFactory } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { throwUnrecoverableError } from '@kbn/task-manager-plugin/server';
+
+import type { UsageRecord } from '../../types';
+import { securityUsageReportingService } from '../services';
+
+const SCOPE = ['serverlessSecurity'];
+// const INTERVAL = '5m';
+const TIMEOUT = '1m';
+export const TYPE = 'serverless-security:endpoint-usage-reporting-task';
+export const VERSION = '1.0.0';
+
+// 1 hour
+// const SAMPLE_PERIOD_SECONDS = 3600;
+
+type MeteringCallback = (metringCallbackInput: MeteringCallbackInput) => UsageRecord[];
+
+export interface MeteringCallbackInput {
+  esClient: ElasticsearchClient;
+  lastSuccessfulReport: Date;
+}
+
+export interface CloudSecurityUsageReportingTaskSetupContract {
+  taskTitle: string;
+  meteringCallback: MeteringCallback;
+  logFactory: LoggerFactory;
+  core: CoreSetup;
+  taskManager: TaskManagerSetupContract;
+}
+
+export interface SecurityMetadataTaskStartContract {
+  taskType: string;
+  interval: string;
+  version: string;
+  taskManager: TaskManagerStartContract;
+}
+
+export class SecurityUsageReportingTask {
+  private logger: Logger;
+  private wasStarted: boolean = false;
+
+  constructor(setupContract: CloudSecurityUsageReportingTaskSetupContract) {
+    const { taskTitle, logFactory, core, taskManager, meteringCallback } = setupContract;
+
+    this.logger = logFactory.get(this.getTaskId());
+    taskManager.registerTaskDefinitions({
+      [TYPE]: {
```

Comment:
I think we need `TYPE` to be different per registration

Reply:
right, fixed that.